pkg/controller/persistentvolume/index.go

```diff
@@ -87,15 +89,29 @@ func (pvIndex *persistentVolumeOrderedIndex) findByClaim(claim *api.PersistentVo
 		unboundVolumes := []*api.PersistentVolume{}
 		for _, volume := range volumes {
 			// volume isn't currently bound or pre-bound.
-			if volume.Spec.ClaimRef == nil {
-				unboundVolumes = append(unboundVolumes, volume)
-				continue
-			}
-
 			if isVolumeBoundToClaim(volume, claim) {
```

Comment:
@jsafrane does volumeName get ignored then? A warning should be thrown. What if the user specifies a different volumeName that the one is pre-bound? Which operation takes precedence?